src/dom.js

```diff
@@ -530,6 +530,40 @@ export function getDataParamsFromAttributes(element, opt_computeParamNameFunc,
 }
 
 /**
+ * Returns link element url components as url parameters key-value pairs.
+ * e.g. <a href="https://example.com/index.html?query#hash"
+ * -> {
+ *   clickHostname: example.com,
+ *   clickProtocol: https,
+ *   clickProtocol: /index.html,
+ *   clickQuery: query,
+ *   clickHash: hash,
+ *   clickUrl: https://example.com/index.html?query#hash
+ * }
+ * @param {!Element} element
+ * @param {function(string):string=} opt_computeParamNameFunc to compute the parameters
+ *    name, get passed the camel-case parameter name.
+ * @return {!JsonObject}
+ */
+export function getDataParamsFromLinkUrl(element, opt_computeParamNameFunc) {
+  if (element.tagName.toUpperCase() != 'A') {
+    return dict();
+  }
+  const computeParamNameFunc = opt_computeParamNameFunc || (key => key);
+  const params = dict();
+  params[computeParamNameFunc('clickHostname')] = element.hostname;
```

Comment:
Are `element.hostname` and similar properties supported on Safari and IE? If not, would it make sense to instead parse the URL and then use the URL object to get these values?

Reply:
changed to use url parse library.